app/jinja_filters.py

```diff
@@ -647,22 +682,32 @@ def map_list_collector_config(
                 }
             )
 
-        rows.append(
-            {
-                "rowItems": [
-                    {
-                        "iconType": icon,
-                        "actions": actions,
-                        "rowTitle": item_name,
-                        "id": list_item.get("list_item_id"),
-                        "rowTitleAttributes": {
-                            "data-qa": f"list-item-{index}-label",
-                            "data-list-item-id": list_item.get("list_item_id"),
-                        },
-                    }
-                ]
-            }
-        )
+        row_items = {
+            "iconType": icon,
+            "actions": actions,
+            "id": list_item.get("list_item_id"),
+            "rowTitleAttributes": row_title_attributes,
+        }
+
+        if answer_title:
+            row_items["valueList"] = [{"text": item_name}]
+
+        row_items["rowTitle"] = answer_title or item_name
+
+        rows.append({"rowItems": [row_items]})
+
+        if related_answers:
+            for block in related_answers.get(list_item["list_item_id"]):  # type: ignore
```

Comment:
Can remove this type ignore if we access directly, otherwise I think we agreed to add comments for type ignores for we can't resolve. Similar for the ignore just below.

Reply:
Accessing by key now which solves the problem.